client/a8c-for-agencies/components/form/hoc/with-error-handling.tsx

```diff
@@ -0,0 +1,66 @@
+import emailValidator from 'email-validator';
+import { useTranslate } from 'i18n-calypso';
+import { ComponentType, useCallback, useMemo, useState } from 'react';
+import { isValidUrl } from 'calypso/a8c-for-agencies/sections/partner-directory/utils/tools';
+
+type Checks = 'non-empty' | 'email' | 'url';
+
+type FieldTypes = string | Array< string >;
+
+type ContextProps< T > = {
+	checks?: Checks[] | null;
+	field?: T;
+	error?: string;
+};
+
+function withErrorHandling< T, F extends FieldTypes >(
+	WrappedComponent: ComponentType< T & ContextProps< F > >
+): ComponentType< T & ContextProps< F > > {
+	return ( props ) => {
+		const { checks, field } = props;
+		const [ error, setError ] = useState< string | undefined >( undefined );
+		const newProps = useMemo( () => ( { ...props, error } ), [ props, error ] );
+		const translate = useTranslate();
+
+		const validate = useCallback( () => {
+			if ( ! checks ) {
+				return;
+			}
+
+			for ( const check of checks ) {
```

Comment:
I was thinking instead of putting all validation here. Maybe we can pass a Validator function instead? This way this function will not bloat when we add more validation types.

```
   checks={ [  validateURL, validateEmpty ] }
```